MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -9,67 +9,105 @@ import Foundation
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Tesla",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false),
-        QuizQuestion(
-            image: "Vivarium",
-            text: "Рейтинг этого фильма больше чем 6?",
-            correctAnswer: false)
-    ]
+    //    private let questions = [
+    //        QuizQuestion(
+    //            image: "The Godfather",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "The Dark Knight",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "Kill Bill",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "The Avengers",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "Deadpool",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "The Green Knight",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: true),
+    //        QuizQuestion(
+    //            image: "Old",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false),
+    //        QuizQuestion(
+    //            image: "The Ice Age Adventures of Buck Wild",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false),
+    //        QuizQuestion(
+    //            image: "Tesla",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false),
+    //        QuizQuestion(
+    //            image: "Vivarium",
+    //            text: "Рейтинг этого фильма больше чем 6?",
+    //            correctAnswer: false)
+    //    ]
     
-    init(delegate: QuestionFactoryDelegate? = nil) {
+    // MARK: - Private properties
+    private let moviesLoader: MoviesLoading
+    private var mostPopularMovie = [MostPopularMovie]()
+    
+    // MARK: - Initialization
+    init(delegate: QuestionFactoryDelegate? = nil, moviesLoader: MoviesLoading) {
         self.delegate = delegate
+        self.moviesLoader = moviesLoader
     }
-
 }
 
+// MARK: - QuestionFactoryProtocol
 extension QuestionFactory: QuestionFactoryProtocol {
-
-    func requestNextQuestion() {
-        guard let index = (0...questions.count).randomElement() else {
-            delegate?.didRecievNextQuestion(question: nil)
-            return
+    
+    func loadData() {
+        moviesLoader.loadMovies { result in
```

Comment:
Не смотря на то что в уроке остались примеры, где нет weak-self у первого замыкания, я рекомендую его добавить. 
В данной реализации, конечно, утечек не возникает. Но замыкание потенциально может захватывать сильно контекст. Чтобы уберечь себя от "ошибок в будущем", стоит ослабить ссылку

```suggestion
        moviesLoader.loadMovies { [weak self] result in
```